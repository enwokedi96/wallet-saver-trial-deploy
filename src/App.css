.App {
  text-align: center;
  overflow-x: hidden;
  background-color: antiquewhite;
}

.container-background {
  background-image: url("./pic11.jpg");
  background-repeat: repeat;
  background-attachment: fixed;
  background-size: contain;
  overflow-x: hidden;
  left: -100px;
}
