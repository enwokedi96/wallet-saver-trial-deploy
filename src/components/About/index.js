import React, {useState} from 'react'
import ModalDialog from '../Modal';

function About() {

    // control states for About modal
    const [showAbout, setAboutShow] = useState(false);
    const handleAboutClose = () => setAboutShow(false);
    const handleAboutShow = () => setAboutShow(true);

    return (
        <>
            <div onClick={handleAboutShow}>
                About Us
            </div>
            <ModalDialog modal_title = "About"
                        modal_content = {<div>
                                            <p className="aboutUsText text-justify">At wallet saver, we believe that everyone deserves to have a better shopping experience. Our goal is to present you with a varied shopping options from AliExpress and Amazon online stores, giving you the best of both worlds. Happy Shopping!!</p>
                                    </div>}
                        show={showAbout} 
                        handleClose={handleAboutClose}
            />
        </>
    )
}

export default About;