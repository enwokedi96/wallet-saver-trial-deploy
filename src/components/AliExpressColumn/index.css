.badge{
    color: black !important;
    background-color: rgb(100, 148, 148);
    font-size: 16px;
    padding: 3px;
}

.property-heading{
    color: rgb(51, 46, 39);
}