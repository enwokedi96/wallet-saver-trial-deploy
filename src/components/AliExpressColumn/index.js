import React from "react";
import "./index.css"

function AliExpressColumn (props) {
    return (<ul className="list-group p-2">
        {props.results.map(result => (
            <li className="list-group-item m-2" key={result.item.itemId}>
            <a target="_blank" href={"https://es.aliexpress.com/item/"+result.item.itemId+".html"} >
            <div className="item-heading mb-2">
            <img alt={result.item.itemId} className="img-fluid shrink" src={result.item.image} />
            <h6 className="m-2 item-heading">{result.item.title}</h6>
            </div></a>
            <strong className="property-heading">rating:</strong> <span className="badge badge-info">{`${result.item.averageStarRate}`}</span> <br/>
            <strong className="property-heading">price:</strong> ${result.item.sku.def.promotionPrice} <br/>
            <strong className="property-heading">number of sales:</strong> {result.item.sales} <br/>
            <strong className="property-heading">freeShipping:</strong> {`${result.delivery.freeShipping}`} <br/>
            </li>
        )
        )}
  </ul>
  )
}

export default AliExpressColumn;