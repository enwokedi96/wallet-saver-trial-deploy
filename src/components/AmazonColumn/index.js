import React from "react";

function CalcDiscount(product_original_price, product_price){
    let discount;
    if (typeof product_original_price === 'string' && typeof product_price === 'string'){
         discount = ` (${Math.floor(((parseInt(product_original_price.slice(1)) - 
                            parseInt(product_price.slice(1))) * 100 /
                            parseInt(product_original_price.slice(1))
                            ))}% discount)`;
    }
    else {
         discount = ""; //" (no discount)"
    }
    return discount;
}

function AmazonColumn (props) {
    return (<ul className="list-group p-2">
        {/* real-time search data extractor 
        {props.otherResults.map(result => (
            <li className="list-group-item m-2" key={result.product_id}>
            <a target="_blank" href={result.product_title} >
            <div className="d-flex align-items-center mb-2">
            <img alt={result.product_id} className="img-fluid shrink" src={result.product_photos[0]} />
            <h4 className="m-2 item-heading">{result.product_title}</h4>
            </div></a>
            <strong>rating:</strong> {result.product_rating} <br/>
            <strong>price:</strong> ${result.offer.price+result.offer.tax} <br/>
            <strong>reviews:</strong> <a target="_blank" href={result.product_reviews_page_url}> Click here </a> 
            <strong>|</strong>
            <strong>Reviews:</strong> <a target="_blank" href={result.product_title}> Click here </a> 
            <br/>
            <strong>shipping:</strong> {`${result.offer.shipping}`} <br/>
            {/* <button type="button" class="btn btn-primary" data-bs-toggle="modal" data-bs-target="#exampleModal">
                View Description
            </button>
            <Modal 
                modal_title = "Product Description"
                modal_content = {result.product_description}
            /> 
            </li>*/}

        {/* Amazon Search Result Extractor */}
        {props.otherResults.map(result => (
            <li className="list-group-item m-2 item-content" key={result.asin}>
            <a target="_blank" href={result.product_url} >
            <div className="item-heading mb-2">
            <img alt={result.asin} className="img-fluid shrink" src={result.product_photo} />
            <h6 className="item-heading m-2">{result.product_title}</h6>
            </div></a>
            <strong className="property-heading">rating:</strong> <span className="badge badge-info">{`${result.product_star_rating}`}</span> <br/>
            <strong className="property-heading">price:</strong> {result.product_price + CalcDiscount(result.product_original_price,result.product_price)} <br/>
            <strong className="property-heading">Current Offers:</strong> {result.product_num_offers}<br/>
            <strong className="property-heading">Best Seller?:</strong> {`${result.is_best_seller}`} <br/>
            </li>
        )
        )}
  </ul>
  )
}

export default AmazonColumn;