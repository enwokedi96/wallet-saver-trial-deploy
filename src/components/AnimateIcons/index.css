.svg-aliexpress{
    display: flex;
    justify-content: center;
    align-items: center;
}

.svg-amazon{
    display: flex;
    justify-content: center;
    align-items: center;
}

.svg-aliexpressXL{
    display: flex;
    justify-content: center;
    align-items: center;
    cursor: pointer;
    transform: scale(7.5);
    padding-left: 15px;
}

.svg-amazonXL{
    display: flex;
    justify-content: center;
    align-items: center;
    cursor: pointer;
    transform: scale(7.5);
    padding-left: 15px;
}