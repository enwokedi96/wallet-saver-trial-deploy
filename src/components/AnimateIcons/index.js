import React, { useEffect, useRef } from "react";
import SVGAnimate from "../SVGAnimate";
import "./index.css";
import { gsap } from "gsap";

// for aliexpress icon
function AnimateIcon(props) {

  const app = props.app;

  // ------------------ GSAP --------------------------
  useEffect(() => {
    const ctx = gsap.context(() => {
      // Target the two specific elements we have asigned the animate class
      // # 1
      // gsap.to("[data-animate='rotate']", {
      //   rotation: 360,
      //   repeat: -1,
      //   repeatDelay: 1,
      //   yoyo: true
      // });
      
      // # 2
      gsap.from("[data-animate='move']", {
        x: 50,
        repeat: -1,
        repeatDelay: 1,
        yoyo: true
      })

      gsap.to("[data-animate='move']", {
        x: -50,
        repeat: -1,
        repeatDelay: 1,
        yoyo: true
      });

      // # 3
      gsap.from("[data-animate='reverse-move']", {
        x: -50,
        repeat: -1,
        repeatDelay: 1,
        yoyo: true
      });

      gsap.to("[data-animate='reverse-move']", {
        x: 50,
        repeat: -1,
        repeatDelay: 1,
        yoyo: true
      });
      
      // 4
      gsap.from("[data-animate='flag']", {xPercent: -10}) 
      gsap.to("[data-animate='flag']", { duration: 1.5, xPercent:10, 
              repeat:-1, yoyo: true, ease:'none' });
      
      // 5
      gsap.from("[data-animate='reverse-flag']", {xPercent: 10}) 
      gsap.to("[data-animate='reverse-flag']", { duration: 1.5, xPercent:-10, 
              repeat:-1, yoyo: true, ease:'none' });

    }, );// <- Scope!
    
    return () => ctx.revert();
  }, []);
  // -------------------------------------------
  return (
    <SVGAnimate
      children = {props.children}
      id = {props.id}
      anim = {props.anim}
    /> );
}

// for aliexpress icon(s)
const AnimateAliExpressIcon = () => {
    return <AnimateIcon children = {<img src="/myIcons/aliexpress.svg" alt=""/>}
            id = "aliexpress" anim = "move"/>
        }

const AnimateAliExpressLargeIcon = (props) =>{
  return <AnimateIcon
          children={<img id="aliExpress" className="img-fluid" 
            src="/myIcons/aliexpress-lrg.svg" alt="AliExpress Icon" 
            onClick={props.handleClick} />}
            id = "aliexpressXL" anim = "flag"
        />
}

// for amazon icon(s)
const AnimateAmazonIcon = () => {
  return <AnimateIcon children = {<img src="/myIcons/amazon.svg" alt=""/>}
          id = "amazon" anim = "reverse-move"/>
      }

const AnimateAmazonLargeIcon = (props) =>{
  return <AnimateIcon
          children={<img id="amazon" className="img-fluid" 
            src="/myIcons/amazon-lrg.svg" alt="Amazon Icon" 
            onClick={props.handleClick} />}
            id = "amazonXL" anim = "reverse-flag"
        />
}

export {AnimateAliExpressIcon, 
        AnimateAliExpressLargeIcon,
        AnimateAmazonIcon,
        AnimateAmazonLargeIcon};
