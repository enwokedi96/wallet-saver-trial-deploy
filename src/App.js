import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import './App.css';

import Footer from './components/Footer';
import Nav from './components/Nav';
import FAQs from './components/FAQs';
import About from './components/About';

import HomePage from './pages/HomePage';
import ChooseStore from './pages/StoreChoice';
import SearchOnlineStores from './pages/SearchResultsTable';

import bg from "./pic11.jpg";

function App() {

  return (
    <div  className='App container-background'>  {/*container-background */}
      <Router>
      <Nav />
      {/* Wrap Route elements in a Routes component */}
      <Routes>
        <Route path="/" element={<HomePage />} />
        <Route path="/search/stores" element={<ChooseStore />} />
        <Route path="/search/stores/aliExpress" element={<SearchOnlineStores searchOpt="aliExpress" />} />
        <Route path="/search/stores/amazon" element={<SearchOnlineStores searchOpt="amazon" />} />
        <Route path="/search/stores/compare" element={<SearchOnlineStores searchOpt="compare" /> } />
        <Route path='/about' element={<About/>}></Route>
        <Route path='/faq' element={<FAQs/>}></Route>
      </Routes>
      <Footer />
  </Router>
</div>
  );
}

export default App;
