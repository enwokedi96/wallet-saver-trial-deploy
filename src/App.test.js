import React from 'react';
import App from './App';
import { shallow, mount } from 'enzyme';
// import { render, screen } from '@testing-library/react';

import NavBar from './components/Nav';
import { AnimateAliExpressLargeIcon, AnimateAliExpressIcon, AnimateAmazonIcon, AnimateAmazonLargeIcon } from './components/AnimateIcons';
import SearchForm from './components/SearchTemplate';
import ModalDialog from './components/Modal';

// App imgs load
  describe('<App>', () => {
    let wrapper;

    it('renders without crashing', () => {
      wrapper = shallow(<App />);
    });
  });

// nav renders
  describe('<Navbar>', () => {
    let wrapper;

    it('renders without crashing', () => {
      wrapper = shallow(<NavBar />);
      })

    it('should render the two buttons class "nav-item', () => {
      expect(wrapper.find('.nav-item')).toHaveLength(2);
    });

    })

// on page for individual stores
  describe("<ChooseStore/>", () => {
    
    it('should render large aliExpress icon', () => {
      const icon = shallow(<AnimateAliExpressLargeIcon/>);
      expect(icon.find("img").prop('src')).toBe('/myIcons/aliexpress-lrg.svg');
      expect(icon.find("img").prop('alt')).toBe('AliExpress Icon');
    });

    it('should render Logo', () => {
      const icon = shallow(<AnimateAmazonLargeIcon/>);
      expect(icon.find("img").prop('src')).toBe('/myIcons/amazon-lrg.svg');
      expect(icon.find("img").prop('alt')).toBe('Amazon Icon');
    });
  })

// on page for comparison
describe("<ChooseStore/>", () => {
    
  it('should render large aliExpress icon', () => {
    const icon = shallow(<AnimateAliExpressIcon/>);
    expect(icon.find("img").prop('src')).toBe('/myIcons/aliexpress.svg');
    expect(icon.find("img").prop('alt')).toBe('');
  });

  it('should render Logo', () => {
    const icon = shallow(<AnimateAmazonIcon/>);
    expect(icon.find("img").prop('src')).toBe('/myIcons/amazon.svg');
    expect(icon.find("img").prop('alt')).toBe('');
  });
})
 
// Unit test: when user fills and submits form, that it triggers the right funcs
  describe('<SearchForm/>', () => {
    const handleSubmit = jest.fn();
    const handleChange = jest.fn();
    const wrapper = shallow(
          <SearchForm
            search=""
            handleFormSubmit={handleSubmit}
            handleInputChange={handleChange}
          />
        );

    it('validates submission on button click', () => {
        const submitBtn = wrapper.find('button')
        submitBtn.simulate('click')
        expect(handleSubmit).toHaveBeenCalled();
      });

    it('validates change on user input', () => {
      const inputField = wrapper.find('input')
      inputField.simulate("change", { target: { value: "foo" }})
      expect(handleChange).toHaveBeenCalled();
    });
  })

// test that modals render correctly
describe('<ModalDialog/>',() => {
  let wrapper;
  beforeEach(() => {
      const container = document.createElement("div");
      document.body.appendChild(container);
      wrapper = shallow( <ModalDialog isOpen={true}/> , {attachTo: container});
  });

  it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();

      expect(wrapper.find('ModalHeader')).toHaveLength(1);

      expect(wrapper.find('ModalBody')).toHaveLength(1);
  });

})