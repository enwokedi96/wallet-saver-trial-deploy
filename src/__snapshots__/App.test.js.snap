// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalDialog/> renders correctly 1`] = `ShallowWrapper {}`;
